template.yaml

```diff
@@ -370,6 +723,342 @@ Resources:
+            Action:
+              - "kms:Decrypt"
+              - "kms:ReEncryptTo"
+              - "kms:GenerateDataKey"
+              - "kms:DescribeKey"
+            Resource: "*"
+            Condition:
+              ArnLike:
+                aws:SourceArn:
+                  - !Sub "arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:/${AWS::StackName}/Config/*"
+          - Effect: Allow
+            Principal:
+              Service: lambda.amazonaws.com
+            Action:
+              - "kms:Decrypt"
+            Resource: "*"
+            Condition: !If
+              - IsDev
+              - ArnLike:
+                  aws:SourceArn:
+                    - !Sub "arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function:*-EncryptDataFunction"
+                    - !Sub "arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function:*-DecryptDataFunction"
+              - ArnEquals:
+                  aws:SourceArn:
+                    - !Sub "arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function:${MainStackName}-EncryptDataFunction"
+                    - !Sub "arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function:${MainStackName}-DecryptDataFunction"
+
+  SecretsManagerEncryptionKeyAlias:
+    Type: AWS::KMS::Alias
+    Properties:
+      AliasName: !Sub "alias/${AWS::StackName}-SecretsManagerEncryptionKey"
+      TargetKeyId: !GetAtt SecretsManagerEncryptionKey.Arn
+
+  IPVCoreApiKeySecret:
+    Type: AWS::SecretsManager::Secret
+    Properties:
+      Name: !Sub "/${AWS::StackName}/Config/API/Key/IPVCore"
+      Description: >-
+        Autogenerated API key for the IPV Core usage.
+        N.B. Manually update IpvCoreApiKey's value upon change, does not auto-update!
+      GenerateSecretString:
+        ExcludeLowercase: false
+        ExcludeNumbers: false
+        ExcludePunctuation: true        # avoids handling escaped chars
+        ExcludeUppercase: false
+        IncludeSpace: false             # avoids handling escaped chars
+        RequireEachIncludedType: true
+        PasswordLength: 128
+      KmsKeyId: !GetAtt SecretsManagerEncryptionKey.Arn
+      Tags:
+        - Key: Product
+          Value: !Ref ProductTagValue
+        - Key: System
+          Value: !Ref SystemTagValue
+        - Key: Environment
+          Value: !Ref Environment
+        - Key: Owner
+          Value: !Ref OwnerTagValue
+        - Key: Source
+          Value: !Ref SourceTagValue
+        - Key: Name
+          Value: !Sub "${AWS::StackName}-IPVCoreApiKeySecret"
+
+  IpvCoreUsagePlanApiKey:
+    Type: AWS::ApiGateway::UsagePlanKey
+    Condition: IsDev
+    Properties:
+      KeyId: !Ref IpvCoreApiKey
+      KeyType: API_KEY
+      UsagePlanId: !Ref IpvCoreApiUsagePlan
+
+  IpvCorePrivateApiUsagePlan:
+    Type: AWS::ApiGateway::UsagePlan
+    DependsOn:
+      - PrivateRestApiStage # This resource is created as part of the PrivateRestApi resource
+    Properties:
+      UsagePlanName: "IPV Core"
```

Comment:
Can we include stack name in the name please, it can very confusing in `dev` where there are multiple stacks deployed to see which usage plan belongs to which stack. Also, include `(private)` as this relates to the private API gateway.